.circleci/config.yml

```diff
@@ -11,11 +15,62 @@ executors:
     docker:
     - image: circleci/golang:1.13
 
+commands:
+  publish_image:
+    parameters:
+      registry:
+        description: Registry address
+        type: string
+        default: "docker.io"
+      organization:
+        description: Registry organization
+        type: string
+        default: prom
+      image_name:
+        description: Name of the container image
+        type: string
+      login:
+        description: Environment variable holding the registry login
+        type: env_var_name
+        default: DOCKER_USER
+      password:
+        description: Environment variable holding the registry password
+        type: env_var_name
+        default: DOCKER_PASSWORD
+      dockerfile_path:
+        description: Path to Dockerfile
+        type: string
+        default: ""
+      context_path:
+        description: Path of docker build context
+        type: string
+        default: "."
+    steps:
+    - run:
+        name: Setup environment variables
+        command: |
+          echo "export TAG=`date +%Y%m%d`" >> $BASH_ENV
+          echo "DOCKER_REPO=<< parameters.registry >>/${DOCKER_ORG:-<< parameters.organization >>}" >> $BASH_ENV
+          echo "DOCKER_IMAGE=<< parameters.image_name >>" >> $BASH_ENV
+    - when:
+        condition: <<parameters.dockerfile_path>>
+        steps:
+        - run: docker build -t $DOCKER_REPO/$DOCKER_IMAGE:$TAG -f << parameters.dockerfile_path >> << parameters.context_path >>
```

Comment:
Did you try using the `Makefile.common` docker commands here? setting, `DOCKERFILE_PATH`. `DOCKER_IMAGE_NAME`, `DOCKER_IMAGE_TAG`, `DOCKERBUILD_CONTEXT` in the `Setup environment variables` step should enable the following command to build the image using `make docker` similarly for `make docker-publish` and `make docker-manifest`.

What `DOCKER_IMAGE_TAG` should be is worth discussing, currently it is `YYYYMMDD` like
@nevill mentioned.

cc: @krasi-georgiev

Reply:
`DOCKERFILE_PATH` and `DOCKERBUILD_CONTEXT` were added in https://github.com/prometheus/prometheus/pull/5669

Some usage of `DOCKERBUILD_CONTEXT` was reduced within the prombench repo in https://github.com/prometheus/prombench/pull/316/files

The prombench cli still does make use of it, which can also be made not to use `DOCKERBUILD_CONTEXT` by adding the following,

```Dockerfile
COPY prombench/prombench /usr/bin/prombench
COPY prombench/cmd/prombench/clone.sh /usr/bin/clone.sh
```
to 
https://github.com/prometheus/prombench/blob/8681448850bdc38dcee93d43513cacda710a4d6e/prombench/cmd/prombench/Dockerfile#L12

I've not tested this but it should work, I somehow forgot to add that in #316

Reply:
I see the target `common-docker` in `Makefile.common`. But it appends `linux` in the middle of image name. I think this project will not create any binary running on OS other than linux, or arch other than amd64, this is so trivial.

Reply:
I think it is fine as is, I even perfer that when it is a single command to have it like this instead of going to the makefile to see what is actually run